app/controllers/manageBusinesses/manage/ConfirmReportingMethodSharedController.scala

```diff
@@ -138,7 +159,7 @@ class ConfirmReportingMethodSharedController @Inject()(val manageIncomeSources:
               form = ConfirmReportingMethodForm(changeTo),
               taxYearEndYear = taxYearModel.endYear.toString,
               taxYearStartYear = taxYearModel.startYear.toString,
-              postAction = getPostAction(taxYearModel.toString, changeTo, isAgent, incomeSourceType),
+              postAction = getPostAction(taxYearModel.toString, changeTo, incomeSourceType),
```

Comment:
This is the cause of your error. You need the submitAgent when the user is an agent.